gradle/publish.gradle

```diff
@@ -84,11 +84,19 @@ publishing {
           def runtimeDeps = project.configurations.runtime.dependencies.collectEntries {
             [it.name, it]
           }
+          depMap.putAll(runtimeDeps)
```

Comment:
Modify the comment block on how/why these dependencies are being collected. For publication, are our `implementation` dependencies needed? Explain and justify? Do these all end up on the manifest classpath?

Reply:
I'm also concerned here.  This "black magic" should probably be more closely examined.

Primarily for me, this is piling all context together, assuming that optional in one configuration is optional in all configurations.  I don't know if this warrants some other means of validation to make sure we haven't drifted in our understanding of what is and isn't optional, and that could well be a future ticket.  Just a worry.

Reply:
Talking with @PurelyApplied, we should wait until gradle releases support for optional dependencies (coming soon) and then remove this black magic stuff.

Implementation dependencies are published - they are published as runtime dependencies in the maven pom. So if they are marked optional, they should be optional in the pom.